File: PrimaMateria/PrimaMateria/Classes/DynoTableLabel.m
//
//  DynoTableLabel.m
//  PrimaMateria
//
//  Created by Jerry Porter on 1/5/2011.
//  Copyright 2016 xTrensa. All rights reserved.
//

@interface DynoTableLabel() {
    UIEdgeInsets edgeInsets;
}

@end

@implementation DynoTableLabel

- (instancetype)initWithFrame: (CGRect)aFrame andColor: (UIColor *)aColor {
    if (self = [super initWithFrame: aFrame]) {
        self.backgroundColor = aColor;
        edgeInsets = UIEdgeInsetsMake(0, 5, 0, 5);
    }
    return self;
}

- (void)drawTextInRect: (CGRect)rect {
    [super drawTextInRect: UIEdgeInsetsInsetRect(rect, edgeInsets)];
}

#pragma mark - Memory Management Methods

@end

File: PrimaMateria/PrimaMateria/Classes/DynoTableLabel.swift
//
//  DynoTableLabel.swift
//  PrimaMateria
//
//  Created by Jerry Porter on 4/14/16.
//  Copyright © 2016 xTrensa. All rights reserved.
//

@objc class DynoTableLabel : UILabel {
    private var edgeInsets : UIEdgeInsets?
    
    // MARK: - Initialization Methods
    
    init(frame: CGRect, andColor: UIColor) {
        super.init(frame:frame)
        edgeInsets = UIEdgeInsetsMake(0, 5, 0, 5)
        self.backgroundColor = andColor
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder:aDecoder)!
    }
    
    // MARK: - View Management Methods

    override func drawRect(rect: CGRect) {
        super.drawTextInRect(UIEdgeInsetsInsetRect(rect, self.edgeInsets!))
    }
}
